Cargo.toml

```diff
@@ -200,6 +200,244 @@ fast-runtime = [ "polkadot-cli/fast-runtime" ]
 runtime-metrics = [ "polkadot-cli/runtime-metrics" ]
 pyroscope = ["polkadot-cli/pyroscope"]
 staging-client = ["polkadot-cli/staging-client"]
+[patch."https://github.com/paritytech/substrate"]
```

Comment:
I typically use this approach: 

https://doc.rust-lang.org/cargo/reference/overriding-dependencies.html#paths-overrides

It says buggy, but I've had no issues and it is 10000x easier. 

also, we've made https://crates.io/crates/diener ourselves.